MapboxSearch/sdk/src/main/java/com/mapbox/search/SearchEngineImpl.kt

```diff
@@ -290,13 +281,8 @@ internal class SearchEngineImpl(
                 )
                 completeSearchResultSelection(resolved)
             }
-            else -> {
-                val error = IllegalArgumentException("Unknown suggestion type ${suggestion.javaClass.printableName}")
-                executor.execute {
-                    reportRelease(error)
-                    callback.onError(error)
-                }
-                return SearchRequestTaskImpl.completed()
+            is BaseSearchSuggestion -> {
```

Comment:
There're a few bugs in Kotlin related to sealed classes/interfaces https://youtrack.jetbrains.com/issues?q=%27when%27%20expression%20must%20be%20exhaustive%20with%20sealed%20interface

And if you try to run this code with Kotlin `1.5.31`

```Kt
sealed interface TestSealedInterface
abstract class TestAbstractClass : TestSealedInterface
class A : TestAbstractClass()
class B : TestAbstractClass()

fun foo(i: TestSealedInterface): String {
    return when(i) {
        is A -> "A"
        is B -> "B"
    }
}
```

You'll get `'when' expression must be exhaustive, add necessary 'is TestAbstractClass' branch or 'else' branch instead`.

> Also I think we can remove !is CoreResponseProvider

Good idea, thanks, I'll remove `CoreResponseProvider` because we don't need it.